src/main.rs

```diff
@@ -235,6 +248,13 @@ fn main() {
     let simplify = get_simplification(matches.value_of("simplify").unwrap());
     let simplification = simplify / (height * width);
 
+    let precision: u32 = match matches.value_of("precision") {
```

Comment:
Based on the comment above, we can move this check into `handle_polyline`

Reply:
I parse `precision` here making it `Option<u32>`. Instead in `handle_polyline` an error is thrown if `precision` is `None`. Is it ok for you?